### PR TITLE
[PROD-770] Update farm/stake funtionality

### PR DESCRIPTION
**Pull request by leqiang-deltafi**
Screen Shot 1401-03-24 at 14 27 06

### CODE REVIEW

File: src/views/Stake/Stake_v2.tsx [hunk 1]
```diff
@@ -83,47 +68,100 @@ const Stake = (): ReactElement => {
...
+        quoteTotalAvailable: new BigNumber(0),
+      };
+    }
 
-  // const quoteTotalStaked = useMemo(() => {
-  //   return farmPool
-  //     ? anchorBnToBn(poolConfig.quoteTokenInfo, farmPool.stakedQuoteShare)
-  //     : new BigNumber(0);
-  // }, [farmPool, poolConfig]);
+    let baseTotalAvailable = anchorBnToBn(
```
[the remark this answers is not in the document]
Comment (ypeng-deltafi) [reply src/views/Stake/Stake_v2.tsx:81]: maybe add some comemnts. I feel you want `baseTotalAvailable = baseTotal - all staked in other farms`. Is that correct?

[the remark this answers is not in the document]
Comment (leqiang-deltafi) [reply src/views/Stake/Stake_v2.tsx:81]: There can be multiple farms. if some of the shares is staked to farm A, there will be less available for staking into farm B

[the remark this answers is not in the document]
Comment (ypeng-deltafi) [reply src/views/Stake/Stake_v2.tsx:81]: did you forget to push?

[the remark this answers is not in the document]
Comment (leqiang-deltafi) [reply src/views/Stake/Stake_v2.tsx:85]: local up-to-date when push. just removed card.tsx and pushed again

File: src/views/Stake/Stake_v2.tsx [hunk 2]
```diff
@@ -212,52 +260,70 @@ const Stake = (): ReactElement => {
...
-        stakeViewActions.setBalance({
-          baseBalance: userTotalBase,
-          quoteBalance: userTotalQuote,
-          baseStaked: userBaseStaked,
-          quoteStaked: userQuoteStaked,
-        }),
-      );
-    }
-  }, [dispatch, poolConfig, userBaseStaked, userQuoteStaked, userTotalBase, userTotalQuote]);
+  // useEffect(() => {
```
Comment (ypeng-deltafi) [inline src/views/Stake/Stake_v2.tsx:263]: I think we can just remove this one

**Review by ypeng-deltafi — APPROVED** [review]

File: src/views/Stake/Stake_v2.tsx (hunk 2 above)
Comment (leqiang-deltafi) [reply src/views/Stake/Stake_v2.tsx]: removed

**Pull request merged**